tests/models/decoder_only/vision_language/test_models.py

```diff
@@ -0,0 +1,629 @@
+"""Common tests for testing .generate() functionality for single / multiple
+image, embedding, and video support for different VLMs in vLLM.
+"""
+import os
+from pathlib import PosixPath
+from typing import Optional, Type
+
+import pytest
+from transformers import AutoModelForVision2Seq
+
+from vllm.utils import cuda_device_count_stateless, identity, is_cpu, is_hip
+
+from ....conftest import (IMAGE_ASSETS, HfRunner, VllmRunner, _ImageAssets,
+                          _VideoAssets)
+from ....utils import fork_new_process_for_each_test, get_memory_gb
+from ...utils import check_outputs_equal
+from .vlm_utils import custom_inputs, model_utils
+from .vlm_utils.case_filtering import get_parametrized_options
+from .vlm_utils.core_utils import (run_custom_inputs_test, run_embedding_test,
+                                   run_multi_image_test, run_single_image_test,
+                                   run_video_test)
+from .vlm_utils.mm_test_types import (CustomTestOptions, ImageSizeWrapper,
+                                      VLMTestInfo, VLMTestType)
+
+# This hack is needed for phi3v & paligemma models
+# ROCm Triton FA can run into shared memory issues with these models,
+# use other backends in the meantime
+# FIXME (mattwong, gshtrasb, hongxiayan)
+if is_hip():
+    os.environ["VLLM_USE_TRITON_FLASH_ATTN"] = "0"
+
+COMMON_BROADCAST_SETTINGS = {
+    "test_type": VLMTestType.IMAGE,
+    "dtype": "half",
+    "max_tokens": 5,
+    "tensor_parallel_size": 2,
+    "image_size_factors": ((.25, 0.5, 1.0), ),
```

Comment:
```suggestion
    "image_size_factors": [(.25, 0.5, 1.0)],
```

Reply:
Using different bracket types helps make it more readable. Similar comment for the image sizes below.